Gems/PhysX/Code/Tests/ShapeColliderComponentTests.cpp

```diff
@@ -566,4 +601,145 @@ namespace PhysXEditorTests
         EXPECT_THAT(aabb.GetMin(), UnitTest::IsClose(-0.5f * boxDimensions * parentScale));
     }
 
+    class PhysXEditorParamBoolFixture
+        : public ::testing::WithParamInterface<bool>
+        , public PhysXEditorFixture
+    {
+    };
+
+    TEST_P(PhysXEditorParamBoolFixture, EditorShapeColliderComponent_ShapeColliderWithQuadShapeNonUniformlyScalesCorrectly)
+    {
+        // test both single and double-sided quad colliders
+        bool singleSided = GetParam();
+
+        EntityPtr editorEntity = CreateInactiveEditorEntity("QuadEntity");
+        editorEntity->CreateComponent(LmbrCentral::EditorQuadShapeComponentTypeId);
+        auto* shapeColliderComponent = editorEntity->CreateComponent<PhysX::EditorShapeColliderComponent>();
+        SetSingleSided(shapeColliderComponent, singleSided);
+        editorEntity->CreateComponent<AzToolsFramework::Components::EditorNonUniformScaleComponent>();
+        const auto entityId = editorEntity->GetId();
+
+        editorEntity->Activate();
+
+        LmbrCentral::QuadShapeComponentRequestBus::Event(entityId, &LmbrCentral::QuadShapeComponentRequests::SetQuadWidth, 1.2f);
+        LmbrCentral::QuadShapeComponentRequestBus::Event(entityId, &LmbrCentral::QuadShapeComponentRequests::SetQuadHeight, 0.8f);
+
+        // update the transform scale and non-uniform scale
+        AZ::TransformBus::Event(entityId, &AZ::TransformBus::Events::SetLocalUniformScale, 3.0f);
+        AZ::NonUniformScaleRequestBus::Event(entityId, &AZ::NonUniformScaleRequests::SetScale, AZ::Vector3(1.5f, 0.5f, 1.0f));
+
+        // make a game entity and check that its AABB is as expected
+        EntityPtr gameEntity = CreateActiveGameEntityFromEditorEntity(editorEntity.get());
+        AZ::Aabb aabb = gameEntity->FindComponent<PhysX::StaticRigidBodyComponent>()->GetAabb();
+
+        EXPECT_NEAR(aabb.GetMin().GetX(), -2.7f, 1e-3f);
+        EXPECT_NEAR(aabb.GetMin().GetY(), -0.6f, 1e-3f);
+        EXPECT_NEAR(aabb.GetMax().GetX(), 2.7f, 1e-3f);
+        EXPECT_NEAR(aabb.GetMax().GetY(), 0.6f, 1e-3f);
```

Comment:
suggestion: If there are many more of these it might be worth writing a matcher for `Aabb` 🙂